File: discover_ruby/manipulate_the_data.rb
require 'HTTPClient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 63d7b0ceea3a07803ac4e808caaa08d51e619bac'
}

clnt = HTTPClient.new
response = clnt.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', query = nil, extheader = {})

my_hash = JSON.parse(response) #stored parsed response in variable my_hash
array_value = my_hash["items"].map { |e| e ["full_name"]} #stored array value in var array_value
puts array_value.join("\n") #prints array_value in format join and next line

File: discover_more_python/manipulate_the_data.py
import urllib2
import json

request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token 27aa992fd7186938f589f8b6da49a7264dcc808a'
}

githuburl = 'https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc'

req = urllib2.Request(githuburl, headers=request_headers)
response = urllib2.urlopen(req) # Call is made

parsed_json = json.loads(response.read()) # Parsed JSON into a Python dictionary 

for i in range(0, 30): # Set for loop to display full names of the 30 most popular projects
   print parsed_json['items'][i]['full_name'] 
